chcemvediet/apps/__init__.py

```diff
@@ -0,0 +1,25 @@
+import datetime
+
+import django.db.backends.sqlite3.base
+from django.apps import AppConfig
+from django.db.backends import utils as backend_utils
+
+
+class ChcemvedietConfig(AppConfig):
+    name = u'chcemvediet.apps'
+
+    def ready(self):
+        django.db.backends.sqlite3.base._sqlite_format_dtdelta = self._sqlite_format_dtdelta
+
+    # fixed in django 3.1rc1
+    def _sqlite_format_dtdelta(self, dt, conn, days, secs, usecs):
+        try:
+            dt = backend_utils.typecast_timestamp(dt)
+            delta = datetime.timedelta(int(days), int(secs), int(usecs))
+            if conn.strip() == u'+':
+                dt = dt + delta
+            else:
+                dt = dt - delta
```

Comment:
Tak ako vzdy doteraz, **nikdy necopypastujeme kod z kniznic**. Namiesto toho v opravenej funkcii zavolame povodnu funkciu, ktorej vysledok pred vratenim fixneme:

```
def ready(self):
    original_sqlite_format_dtdelta = django.db.backends.sqlite3.base._sqlite_format_dtdelta
    def new_sqlite_format_dtdelta(*args):
        res = original_sqlite_format_dtdelta(*args)
        return res.strip(u'+00:00')
    django.db.backends.sqlite3.base._sqlite_format_dtdelta = new_sqlite_format_dtdelta
```

Toto je len zhruba napisane zhlavy. Este to bude treba doladit. Napriklad, ak povodna funkcia vrati `None`.

Reply:
:ok: